client/reader/recent/index.tsx

```diff
@@ -18,31 +20,31 @@ import './style.scss';
 const Recent = () => {
 	const dispatch = useDispatch< ThunkDispatch< AppState, void, AnyAction > >();
 	const [ selectedItem, setSelectedItem ] = useState< ReaderPost | null >( null );
+	const isWide = useBreakpoint( WIDE_BREAKPOINT );
 
 	const [ view, setView ] = useState< View >( {
 		type: 'table',
 		fields: [ 'seen', 'post' ],
 	} );
 
-	const { data, posts } = useSelector( ( state: AppState ) => {
-		const streamData = state.reader?.streams?.recent;
-		const postsMap: Record< string, PostItem > = {};
+	const data = useSelector( ( state: AppState ) => state.reader?.streams?.recent );
 
-		// Create a map of posts for all items
-		streamData?.items?.forEach( ( item: ReaderPost ) => {
+	const posts = useSelector( ( state: AppState ) => {
+		const items = data?.items;
+		if ( ! items ) {
+			return {};
+		}
+
+		return items.reduce( ( acc: Record< string, PostItem >, item: ReaderPost ) => {
 			const post = getPostByKey( state, {
-				feedId: +item.blogId,
-				postId: +item.postId,
+				feedId: item.blogId,
```

Comment:
I removed the conversion to a number as unnecessary (we get a number from the store).